### DISCUSSION TITLE
support for hasOnlyDefaultVariant on Product object

### DISCUSSION THREAD

**Discussion by NathanPJF** [Shopify/storefront-api-feedback#117] [Ideas & Requests]
Request for the [Product object](https://shopify.dev/api/storefront/2022-07/objects/Product) of Storefront API to have the [`hasOnlyDefaultVariant`](https://shopify.dev/api/admin-graphql/2022-07/objects/Product#field-product-hasonlydefaultvariant) prop that is available in the Admin API.

In the Hyrdogen repo (https://github.com/Shopify/hydrogen/issues/1761) I pointed out that the demo store was unnecessarily showing the variant option name and option value. I suggested they reference the `hasOnlyDefaultVariant` prop to conditionally display this information, but I was told it was available in the Storefront API.

<details><summary>Example screenshot of a product that doesn't have any merchant-defined product options</summary>

![](https://screenshot.click/27-25-2o6vj-eg1uc.png)

</details>

My recommendation was based on the Dawn theme's Liquid template: https://github.com/Shopify/dawn/blob/21468abc2eb83632fde8c9c5ae497218520f1d66/sections/main-cart-items.liquid#L110-L117

## Other details

I was asked if "hasOnlyDefaultVariant" could this be emulated by checking to see how many variants there are for a product? The answer is : Yes and no.

* Yes: If the store designer feels it's not useful to show the product option information to Buyers. One could say "if variant size is 1, don't show product option info" 
* No: Because maybe the store owner wants to let people know the variant `Size` is `M` for this product, even if it's the only one option and value because all other fashion products have this UI on the product page. So basing the display off of the variant count hides the information.

**Comment by yuzushioh** [comment]
This would be a big help for me too! 🙏🏻
